mne/encoding/model.py

```diff
@@ -91,31 +116,37 @@ def fit(self, X, y=None):
         n_features = X.shape[0]
 
         # Fit the model and assign coefficients
-        self.coef_ = np.zeros([n_chs, n_features])
-        for ii in range(n_chs):
-            i_y = y[ii][:, np.newaxis]
-            self.est.fit(X.T, i_y)
-            self.coef_[ii, :] = self.est.steps[-1][-1].coef_
+        self.est.fit(X.T, y.T)
+        self.coef_ = self.est._final_estimator.coef_
         self.X = X
         self.y = y
 
     def predict(self, X, preproc_x=None):
         """Generate predictions using fit coefficients.
 
-        This uses the `coef_` attribute for predictions.
+        This uses the predict method of the final estimator in the
+        `est` attribute.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input feature array.
         """
+        # Preprocess X with the pipeline
         if preproc_x is not None:
             X = preproc_x.fit_transform(X)
-        X = self.est._pre_transform(X.T)[0].T
-        preds = np.dot(self.coef_, X)
-        return preds
+        return self.est.predict(X.T)
 
 
 def _check_estimator(est):
     from sklearn.linear_model import Ridge
     from sklearn.pipeline import Pipeline
     if est is None:
         est = Ridge()
+    if not isinstance(est, str) and not is_regressor(est):
```

Comment:
remove `not isinstance(est, str)` and move this test after L. 150 with an `elif`

BTW, the string should initialize a sklearn-like estimator so that you can do

```
if isinstance(est, str):
    est = get_solver(est)

if not isregressor(est):
   raise ValueError()
```